### ISSUE TITLE
Canonical fork / gem maintenance

### ISSUE THREAD

**Issue by halo** [digineo/xrechnung#2]
Hi! First of all, thank you for this gem. It's definitely saving me a lot of work :)

I noticed that there are various forks [such as this one](https://github.com/digineo/xrechnung/compare/eba86880495848ddaf2f713f655a5a39b5faeb5b...projo-berlin:xrechnung:fd2e945705565aa3860e5f1d750a7f5459fa68c9) which added a number of new XML-tags (which I also need). But there was no concerted merging back into your original code, not to mention the gem on RubyGems.

I wanted to ask if you're generally open to pull requests if they are in an ordered manner? Would you then be willing to push the changes to RubyGems on a regular basis? Maybe something [like in this manner](https://github.com/halo/xrechnung/pulls) where I backported the changes (and fixed the specs).

Also, if I understand it correctly, the validator is only used to [test some fixtures](https://github.com/digineo/xrechnung/blob/eba86880495848ddaf2f713f655a5a39b5faeb5b/Rakefile#L48). It would be great if the code in the test suite could generate various complete xRechnung documents (a minimal one, one with all possible tags, one with some optional tags) and then have the validator check all the generated code. This way, the test suite would actually catch XML files which this gem generated but are invalid. I have a [proof-of-concept for that](https://github.com/halo/xrechnung/pull/1) 

I'd be grateful for your thoughts and thank you for your time. Let me maybe also ping @florian2 into the boat for feedback. 

PS: How important is it for you to remain on Ruby `2.7` (which is end of life)? 

PPS: You have a *heavily* customized Rubocop file. In my experience, open source collaboration is greatly improved when the linter is as close to the convention as possible. Are you open to getting closer towards standard Rubocop?

PPPS: I'm not sure about the `# noinspection RubyResolve` comments, since I'm not using IntelliJ. I take it, the YARD `@!attribute` comments are auto-generated? The "noinspection" shows up in [the documentation](https://rubydoc.info/gems/xrechnung/Xrechnung/PartyLegalEntity) at any rate.

**Comment by florian2** [comment]
@halo I am here and available for a zoom call to discuss things further. please write an email to info@projo.berlin and I get back to you?

**Comment by florian2** [comment]
@dmke @corny pinging you directly. Can you please comment on what's the state of the gem and who is maintaining it?

**Comment by dmke** [comment]
Hi,

Sorry for the delayed answer, I've reduced the number of notifications Github sends to me (that also meant that I don't immediately see new issues).

We're generally interested in PRs (of any form, really), but availability for review/merging is extremely limited at the moment. I'm not actually sure whether @corny or @tg-digineo maintains this code (I'm not directly involved in its development), but I can give you some bullet points:

- Last time I checked, this library covers the cases we needed for our application(s). This is not to say that we miss required schema upgrades - our customers have yet to complain (or better: not pay) ;-)
- We do still need 2.7-compatability because one application cannot easily be upgraded (though that's a work-item I need to address until the end of the year...)
- Regarding Rubocop: That's what we use across all our Ruby projects, we obviously don't agree what "standard Ruby" should look like. It should not affect contributions, because `rubocop --safe --autocorrect` exists.
- The IntelliJ/RubyMine comments are legacy, my colleague has switched back to (Neo-) Vim.
